### DISCUSSION TITLE
No need for this line "test_pred = torch.softmax(test_logits, dim=1).argmax(dim=1)"  in traning loop for the model

### DISCUSSION THREAD

**Discussion by aryamahadik** [mrdbourke/pytorch-deep-learning#1044] [Ideas]
As the work of softmax function to classify the output so there is no need to add "test_pred = torch.softmax(test_logits, dim=1).argmax(dim=1)" this line in traning loop, nothing to do with this line in the loop! if iam wrong pls corret me.

**Comment by mrdbourke** [comment]
Hey @aryamahadik ,

You are correct, there is no *need* to add that line: 

```
test_pred = torch.softmax(test_logits, dim=1).argmax(dim=1)
```

In the testing loop.

You could do the same with: 

```
test_pred = test_logits.argmax(dim=1)
```

This will give you the same result.

However, if you want prediction probabilities, you can use:

```
pred_probs = torch.softmax(test_logits, dim=1)
```

**Reply by aryamahadik** [reply]
thank you bro to letting me know this i got confused in that line, bro your pytorch course is really helping me thanku for this quality content ! lots of love !
